### PR TITLE
Add a RSR convert script for FY-4B GHI sensor

### PR DESCRIPTION
**Pull request by yukaribbba**
- [x] Add your name to `AUTHORS.md` if not there already

Original RSR comes from CMA's website. https://img.nsmc.org.cn/PORTAL/NSMC/DATASERVICE/SRF/FY4B/FY4B_GHI_SRF.zip

The hdf result. See if we can put it on zenodo:
[rsr_ghi_FY-4B.zip](https://github.com/pytroll/pyspectral/files/12509225/rsr_ghi_FY-4B.zip)

The images:
(rayleigh correction)
image

(no correction)
image

You see the correction is not that satisfying. I think it's because the original rsr data is kinda rough. But anyway that's the only open data I can find on CMA's website.

### CODE REVIEW

**Comment by simonrp84** [comment]
Hi @yukaribbba, I can't find the GHI RSRs on the NSMC website any more - your link is no longer working and I can't find an alternative (only the AGRI RSRs). Do you know where do download these now?

(only just realised I was requested to review this - doing so now!)

File: rsr_convert_scripts/ghi_rsr.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2018-2022 Pytroll developers
```
Comment (simonrp84) [inline rsr_convert_scripts/ghi_rsr.py:4]: This should be `2018-2023`

File: rsr_convert_scripts/ghi_rsr.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Read the FY-4B GHI relative spectral responses.
+
+Data from http://satellite.nsmc.org.cn/PortalSite/StaticContent/FileDownload.aspx?CategoryID=1&LinkID=687
```
Comment (simonrp84) [inline rsr_convert_scripts/ghi_rsr.py:22]: The link doesn't seem to work any more. This lets me download it though:
`http://satellite.nsmc.org.cn/PortalSite/StaticContent/FileDownload.aspx?currentculture=zh-CN&CategoryID=1&LinkID=687`

Comment (yukaribbba) [reply rsr_convert_scripts/ghi_rsr.py:22]: Well, looks like they just changed http to https. I thought it was a new version at first look....

**Comment by yukaribbba** [comment]
@simonrp84 The works is done but not perfect.
I just found a GHI image from a NSMC promotion document on JMA website. 
image

The color is pretty nice here. Looks like MODIS/MERSI-2. I don't know what has happened. Maybe the performance has degraded?

**Comment by simonrp84** [comment]
@yukaribbba To my knowledge the instrument hasn't degraded, the images have looked like this right from the start. I presume that a lot of post-processing was done to make their image look as it does.

Also, I just tried to run your conversion script and am getting some errors:
```
(base) C:\Users\Simon\PycharmProjects\pyspectral\rsr_convert_scripts>python ghi_rsr.py
Traceback (most recent call last):
  File "C:\Users\Simon\PycharmProjects\pyspectral\rsr_convert_scripts\ghi_rsr.py", line 113, in <module>
    convert_ghi()
  File "C:\Users\Simon\PycharmProjects\pyspectral\rsr_convert_scripts\ghi_rsr.py", line 106, in convert_ghi
    tohdf5(GHIRSR, 'FY-4B', FY4_GHI_BAND_NAMES)
  File "C:\Users\Simon\PycharmProjects\pyspectral\pyspectral\utils.py", line 274, in convert2hdf5
    instr = ClassIn(bandnames[0], platform_name)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Simon\PycharmProjects\pyspectral\rsr_convert_scripts\ghi_rsr.py", line 56, in __init__
    self._get_bandfilenames()
  File "C:\Users\Simon\PycharmProjects\pyspectral\pyspectral\raw_reader.py", line 66, in _get_bandfilenames
    self.options[self.platform_name + '-' +
KeyError: 'ch1'
```

I added this to `pyspectral.yaml` before running:
```
FY-4B-ghi:
  path: C:/Users/Simon/Downloads/FY4B_GHI_SRF/
```
Any idea what might be happening there?

**Comment by yukaribbba** [comment]
Nothing wrong with my yaml:

```
FY-4B-ghi:
    path: D:/satpy_config/FY4B_GHI_SRF
    ch1: FY4B_GHI_SRF_CH01_Pub.txt
    ch2: FY4B_GHI_SRF_CH02_Pub.txt
    ch3: FY4B_GHI_SRF_CH03_Pub.txt
    ch4: FY4B_GHI_SRF_CH04_Pub.txt
    ch5: FY4B_GHI_SRF_CH05_Pub.txt
    ch6: FY4B_GHI_SRF_CH06_Pub.txt
    ch7: FY4B_GHI_SRF_CH07_Pub.txt
```

**Comment by simonrp84** [comment]
Yes, you need to include that YAML in the PR :)

**Comment by yukaribbba** [comment]
oops...

**Comment by adybbroe** [comment]
@yukaribbba Can you just please fix the pre-commit CI complaints?
Then I will prepare data on Zenodo

**Comment by yukaribbba** [comment]
Hi @adybbroe thanks for reviewing! The link is available for me. You might want to try it several times since their website is unstable or hard to be connected sometimes. Here's the original RSR data pulled from that link.
[FY4B_GHI_SRF.zip](https://github.com/pytroll/pyspectral/files/13487761/FY4B_GHI_SRF.zip)

**Comment by yukaribbba** [comment]
I remember there's a way to call pre-commit CI and let it do the fix automatically but I forgot how to do that....

**Comment by yukaribbba** [comment]
@adybbroe Ok done.

**Comment by mraspaud** [comment]
@adybbroe could you have another look at this to see if we can merge this?

**Comment by yukaribbba** [comment]
@adybbroe could you take a moment to see if there's anything else to do for this PR?

**Comment by yukaribbba** [comment]
I've updated the link to original rsr in description and it's valid now.
[https://img.nsmc.org.cn/PORTAL/NSMC/DATASERVICE/SRF/FY4B/FY4B_GHI_SRF.zip](url)

**Review by adybbroe — APPROVED** [review]
LGTM, thanks! And sorry for the delay.

**Comment by yukaribbba** [comment]
Thanks! :D

**Pull request merged**